testing/integ-test/src/integTest/groovy/org/gradle/api/tasks/diagnostics/FullTaskListingTaskReportTaskIntegrationTest.groovy

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.diagnostics
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class FullTaskListingTaskReportTaskIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
Why does this live in the `integ-test` subproject and not diagnostics base?

Reply:
The full task listing wouldn't be available in the distribution used by `diagnostics-base`.